grid2op/tests/test_EpisodeData.py

```diff
@@ -137,6 +146,41 @@ def test_3_episode_3process_with_saving(self):
             assert np.abs(
                 dt_float(episode_data.meta["cumulative_reward"]) - self.real_reward) <= self.tol_one
 
+    def test_with_opponent(self):
+        init_budget = 1000
+        opponent_attack_duration = 15
+        opponent_attack_cooldown = 30
+        opponent_budget_per_ts = 0.
+        opponent_action_class = TopologyAction
+
+        LINES_ATTACKED = ["1_3_3", "1_4_4", "3_6_15", "9_10_12", "11_12_13", "12_13_14"]
+
+        p = Parameters()
+        p.NO_OVERFLOW_DISCONNECTION = True
+        env = make("rte_case14_realistic",
+                   test=True, param=p,
+                   opponent_init_budget=init_budget,
+                   opponent_budget_per_ts=opponent_budget_per_ts,
+                   opponent_attack_cooldown=opponent_attack_cooldown,
+                   opponent_attack_duration=opponent_attack_duration,
+                   opponent_action_class=opponent_action_class,
+                   opponent_budget_class=BaseActionBudget,
+                   opponent_class=RandomLineOpponent,
+                   kwargs_opponent={
+                       "lines_attacked": LINES_ATTACKED
+                   })
+        env.seed(0)
+        runner = Runner(**env.get_params_for_runner())
+
+        f = tempfile.mkdtemp()
+        res = runner.run(nb_episode=1,
+                         env_seeds=[4], agent_seeds=[0],
+                         max_iter=opponent_attack_cooldown - 1,
+                         path_save=f)
+
+        episode_data = EpisodeData.from_disk(agent_path=f, name=res[0][1])
+        assert episode_data.attacks[0].as_dict()['set_line_status']['disconnected_id'][0] == 3
```

Comment:
I don't understand.

Can you used:
lines_impacted, subs_impacted = attack.get_topological_impact()
and check that lines_impacted[3] is True
instead ?

This would be more explicit i believe.

Reply:
Done in a010bc3e.